petastorm/reader.py

```diff
@@ -607,7 +646,10 @@ def __next__(self):
                                'make_reader/make_batch_reader context manager has exited but you try to '
                                'fetch a sample from it anyway')
         try:
-            return self._results_queue_reader.read_next(self._workers_pool, self.schema, self.ngram)
+            results = self._results_queue_reader.read_next(self._workers_pool, self.ngram)
+            if '_schema' in self.__dict__:
```

Comment:
It might be better to (a) check if the Transform_spec.func returned proper types right after we run it on a worker thread/process and (b) only in case that we know that we work with Transform_spec and not a callable while we raise errors and not warnings. My reasoning is:
(a): Main thread/process is an expensive place to run the code - running it on the worker leaves more opportunity for parallelization.
(b): Messing up types with TransformSpec is a big deal - user promissed us something and violated it, hence "deserves" to crash. In the "transform is callable"  case, user has a carte blache to do whatever they want with the types, so warning seems inappropriate.